### ISSUE TITLE
Huge delay on opening Camera - UWP

### ISSUE THREAD

**Issue by himion0** [jamesmontemagno/Xamarin.Plugins#217]
Hi,

I've essentially been using this to try and open up a camera page. Its however taking over a minute to actually open the camera page. I'm using xamarin forms and deploying to the windows 10 UWP platform. 

```
using Plugin.Media;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Xamarin.Forms;

namespace Vigour.contentpages
{
    public partial class CameraPage : ContentPage
    {
        public CameraPage()
        {
            InitializeComponent();
        }
        
        private void print(String s)
        {
            Debug.WriteLine(s);
        }

        // when the button in the xaml file with parameter Clicked="buttonClicked" is clicked,
        // the method is called. 
        private async void buttonClicked(object sender, EventArgs e)
        {

            print(CrossMedia.Current.IsCameraAvailable.ToString());

            if (!CrossMedia.Current.IsCameraAvailable || !CrossMedia.Current.IsTakePhotoSupported)
            {
                print("Camera is availible");
                await DisplayAlert("No Camera", ":( No camera available.", "OK");
                return;
            }

            print("Start taking photo");
            var file = await CrossMedia.Current.TakePhotoAsync(new Plugin.Media.Abstractions.StoreCameraMediaOptions
            {
                Directory = "Sample",
                Name = "test.jpg"
            });

            print("Saved photo");
            if (file == null)
                return;



            await DisplayAlert("File Location", file.Path, "OK");

            image.Source = ImageSource.FromStream(() =>
            {
                var stream = file.GetStream();
                file.Dispose();
                return stream;
            });
        }
    }


}
```

**Comment by jamesmontemagno** [comment]
On Windows 10 mobile or Windows 10 desktop?

On desktop it comes up in a pop up and opens for me in under a second.

**Comment by himion0** [comment]
Windows 10 Phone, Lumia 640

**Comment by jamesmontemagno** [comment]
I have the 635 and will test out today :)

On Wed, Feb 10, 2016 at 8:38 AM, Harry Blum <notifications@github.com>
wrote:

> Windows 10 Phone, Lumia 640
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jamesmontemagno/Xamarin.Plugins/issues/217#issuecomment-182468655>
> .
>



-- 
James Montemagno
http://www.montemagno.com
@jamesmontemagno <http://www.twitter.com/jamesmontemagno>

**Comment by himion0** [comment]
Great thanks james!

**Comment by himion0** [comment]
Did you manage to test it?

**Comment by jamesmontemagno** [comment]
Just test it and it takes about 2 seconds for the camera app to launch and another 4 or 5 to actually show the finder. It is a slow phone of course. Just using standard apis. All I am doing is TakePhotoAsync.

What build are you on?

I am on Version 1511
OS Build 10.0.10586.71

**Comment by Savier9** [comment]
Hi guys, this delay is caused by CrossMedia.Current.IsCameraAvailable. Once removed, camera shows up within secs - but bad idea to remove it, I guess. This also aplies only to UWP and W81.

**Comment by jamesmontemagno** [comment]
@Savier9  that could not possibly be the issue about anything in XAML parsing because W81 and UWP do not use anything custom at all: https://github.com/jamesmontemagno/Xamarin.Plugins/blob/master/Media/Media/Media.Plugin.WindowsStore/MediaImplementation.cs the code is there.

I also just tested it on a lumia 635 and my samsung machine and it worked just fine within seconds.

Which did you remove as far as code goes. Do you have a better example?

**Comment by jamesmontemagno** [comment]
Ahhh @Savier9 I assume this code looks proplemenatic:
```
  print(CrossMedia.Current.IsCameraAvailable.ToString());

            if (!CrossMedia.Current.IsCameraAvailable || !CrossMedia.Current.IsTakePhotoSupported)
            {

``

Can you try do add the Initailization first:

```
await CrossMedia.Current.Initialize();
  print(CrossMedia.Current.IsCameraAvailable.ToString());

            if (!CrossMedia.Current.IsCameraAvailable || !CrossMedia.Current.IsTakePhotoSupported)
            {
```

**Comment by Savier9** [comment]
Yeah, thats it, init solved the issue :+1:

**Comment by jamesmontemagno** [comment]
Nice, I will document and figure out what is going on there. Probably a deadlock of some sort. Thanks!

**Comment by Savier9** [comment]
I forgot to say, that Im using windows emulators. I have encountered this error earlier in VS2012, where assembly names couldnt contain dots, but that was fixed.

The parsing error comes from this XAML https://github.com/jamesmontemagno/Xamarin.Plugins/blob/master/Media/Media/Media.Plugin.WindowsPhone81/CameraCaptureUI.xaml

and the same error was thrown with UWP and 81 RT

I have validated the xaml and there is no typo. So I just referenced all projects manually and parsing error was gone. That makes me think something suspicious is within the nuget package.
